### PR TITLE
Remove ffmpeg-based RTP receiver in favor of the rust `rtp` crate

### PR DESCRIPTION
**Pull request by jerzywilczek**

### CODE REVIEW

**Review by WojciechBarczynski — APPROVED** [review]

File: src/rtp_receiver.rs [hunk 1]
```diff
@@ -39,228 +26,120 @@ pub struct Options {
...
+            .await;
+
+            match received_bytes {
+                Some(n) => {
+                    let packet = match rtp::packet::Packet::unmarshal(&mut &buffer[..n]) {
+                        // https://datatracker.ietf.org/doc/html/rfc5761#section-4
+                        //  Given these constraints, it is RECOMMENDED to follow the guidelines
+                        //  in the RTP/AVP profile [7] for the choice of RTP payload type values,
+                        //  with the additional restriction that payload type values in the range
+                        //  64-95 MUST NOT be used.
```
Comment (wkozyra95) [inline src/rtp_receiver.rs:101]:
```suggestion
                        // https://datatracker.ietf.org/doc/html/rfc5761#section-4
                        // Given these constraints, it is RECOMMENDED to follow the guidelines
                        // in the RTP/AVP profile [7] for the choice of RTP payload type values,
                        // with the additional restriction that payload type values in the range
                        // 64-95 MUST NOT be used.
```

File: src/rtp_receiver.rs [hunk 2]
```diff
@@ -39,228 +26,120 @@ pub struct Options {
...
+                    None
+                },
+                async {
+                    let (len, _) = socket.recv_from(&mut buffer).await.unwrap();
+                    Some(len)
+                },
+            )
+            .await;
+
+            match received_bytes {
```
Comment (wkozyra95) [inline src/rtp_receiver.rs:94]:
```suggestion
            let Some(received_bytes) = received_bytes else {
                return
            }
```

clearer intention + less indentation in the rest of the code.

technically you could do break, but I think return shows intention better

File: compositor_pipeline/src/pipeline/decoder.rs [hunk 3]
```diff
@@ -48,9 +50,18 @@ impl<Input: PipelineInput> Decoder<Input> {
         let parameters = input.decoder_parameters();
 
         std::thread::spawn(move || {
+            let mut h264_depayloader = rtp::codecs::h264::H264Packet::default();
```
Comment (wkozyra95) [inline compositor_pipeline/src/pipeline/decoder.rs:53]: I don't like the fact that we are using here anything from rtp package. Packets that we receive here should already in a format that does not depend on any protocol

File: examples/web_view.rs [hunk 4]
```diff
@@ -152,6 +152,8 @@ fn start_example_client_code() -> Result<()> {
             "-an",
             "-c:v",
             "libx264",
+            "-profile:v",
```
Comment (wkozyra95) [inline examples/web_view.rs:155]: verify if this works without this

File: src/rtp_receiver.rs [hunk 5]
```diff
@@ -39,228 +26,120 @@ pub struct Options {
...
-                decoder_params_sender
-                    .send(Err(Box::new(InitError::FfmpegError(e))))
-                    .unwrap();
-                return;
-            }
-        };
+        loop {
+            let received_bytes = future::or(
+                async {
+                    should_close_rx.recv().await.unwrap();
```
[the remark this answers is not in the document]
Comment (jerzywilczek) [reply src/rtp_receiver.rs:84]: It only errors if the other side of the channel is closed, which wont happen

File: src/rtp_receiver.rs (hunk 2 above)
Comment (jerzywilczek) [reply src/rtp_receiver.rs:94]: I want to address this in the following PRs, so I'm leaving this as-is for now.

File: compositor_pipeline/src/pipeline/decoder.rs (hunk 3 above)
Comment (jerzywilczek) [reply compositor_pipeline/src/pipeline/decoder.rs:53]: I want to address this in the following PRs, so I'm leaving this as-is for now.

File: examples/web_view.rs (hunk 4 above)
Comment (jerzywilczek) [reply examples/web_view.rs:155]: Removing this, committed it accidentally

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**